osquery/core/database/database.cpp

```diff
@@ -0,0 +1,51 @@
+/**
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/core/database/database.h>
+
+namespace osquery {
+
+Expected<int32_t, DatabaseError> Database::getInt32(const std::string& domain, const std::string& key) {
+  Expected<std::string, DatabaseError> string_value = getString(domain, key);
+  if (string_value) {
+    try {
+      return std::stoi(*string_value);
+    }
+    //std::invalid_argument or std::out_of_range
+    catch (std::exception  e) {
+      return createError(DatabaseError::FailToReadData, "Failed to convert string to int");
+    }
+  } else {
+    return string_value.takeError();
+  }
+}
+
+ExpectedSuccess<DatabaseError> Database::putInt32(const std::string& domain, const std::string& key, const int32_t value) {
+  std::string buffer = std::to_string(value);
+  return putString(domain, key, buffer);
+}
+
+Expected<int32_t, DatabaseError> Database::getInt32Or(const std::string& domain, const std::string& key, const int32_t default_value) {
+  auto result = getInt32(domain, key);
+  if (!result && result.getError() == DatabaseError::KeyNotFound) {
+    return default_value;
+  }
+  return result;
+}
+
+Expected<std::string, DatabaseError> Database::getStringOr(const std::string& domain, const std::string& key, const std::string default_value) {
+  auto result = getString(domain, key);
+  if (!result && result.getError() == DatabaseError::KeyNotFound) {
```

Comment:
There is a method `.isValue()` and `.isError()` in Expected, which you can use instead of `operator bool`. To be more explicit, if you want to :)

Reply:
idk, I'm tried booth ways and in my opinion value and !value are easier to read especially in cases when we do 
if (value) {
}

Reply:
As I as said, If you want to ;)